### PR TITLE
#2246 MappedColumn fails: Expected an option type, found Mapped[Optio…

### PR DESCRIPTION
**Pull request by shmishleniy**

### CODE REVIEW

**Comment by nafg** [comment]
@helllamer can you review this?

File: slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala [hunk 1]
```diff
@@ -335,6 +335,7 @@ class AggregateTest extends AsyncTest[RelationalTestDB] {
     val q5c = as.distinct.map(a => (a.id, a.a))
     val q6 = as.distinct.length
     val q7 = as.map(a => (a.a, a.b)).distinct.take(10)
+    val q8 = as.map(a => (a.a, a.b)).distinct.take(10)
```
Comment (helllamer) [inline slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala:338]: Duplicate unused line.

File: slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala [hunk 2]
```diff
@@ -356,32 +357,44 @@ class AggregateTest extends AsyncTest[RelationalTestDB] {
...
+                   , BData("3", "c", "b", Some(StrWrapper("v1")))
                    )
 
-    val qb1a = bs.map(_.a).distinct
+    val qb1a = bs.map(_.a).distinctOn(identity)
     val qb1b = bs.distinctOn(_.a).map(_.a)
     val qb2 = bs.distinctOn(_.a).map(a => (a.a, 5))
     val qb3a = bs.distinctOn(_.a).map(_.id).filter(_ === "1") unionAll bs.distinctOn(_.a).map(_.id).filter(_ === "2")
-    val qb4 = bs.map(a => (a.a, a.b)).distinct.map(_._1)
+    val qb4 = bs.map(a => (a.a, a.b)).distinctOn(a => (a._1, a._2)).map(_._1)
```
Comment (helllamer) [inline slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala:389]: Sorry, i don't understand difference with `.distinctOn(identity)` (possibly, my Slick skills is not enought here).

File: slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala [hunk 3]
```diff
@@ -356,32 +357,44 @@ class AggregateTest extends AsyncTest[RelationalTestDB] {
...
     val qb5a = bs.groupBy(_.a).map(_._2.map(_.id).min.get)
     val qb5b = bs.distinctOn(_.a).map(_.id)
     val qb5c = bs.distinctOn(_.a).map(a => (a.id, a.a))
     val qb6 = bs.distinctOn(_.a).length
-    val qb7 = bs.map(a => (a.a, a.b)).distinct.take(10)
+    val qb7 = bs.map(a => (a.a, a.b)).distinctOn(a => (a._1, a._2)).take(10)
```
Comment (helllamer) [inline slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala:394]: Sorry, i don't understand difference with .distinctOn(identity) (possibly, my Slick skills is not enought here).

**Review by helllamer — CHANGES_REQUESTED** [review]
Looks like, one line need to be removed.
About two `distinctOn(...)` i'm unsure, as wrote in comments below/above.

**Comment by nafg** [comment]
@shmishleniy can you respond to @helllamer 's comments please?

File: slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala (hunk 2 above)
Comment (shmishleniy) [reply slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala:389]: just want to cover all variation, we shouldn't rely on function implementation in tests

File: slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala (hunk 3 above)
Comment (shmishleniy) [reply slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala:394]: same here

File: slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala (hunk 1 above)
Comment (shmishleniy) [reply slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala]: fixed

**Review by helllamer — APPROVED** [review]

**Review by helllamer — APPROVED** [review]

**Review by helllamer — APPROVED** [review]

**Review by nafg — APPROVED** [review]

**Pull request merged**